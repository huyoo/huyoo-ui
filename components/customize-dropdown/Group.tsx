/**
 * @DECS: 下拉，已选择的群组
 * @AUTH: hy
 * @DATE: 2020-09-23
 */
import React from "react";
import {Avatar, List} from "antd";
import {CloseOutlined} from "@ant-design/icons/lib";
import DropdownBox from "./DropdownBox";

export interface GroupProp {
  selectedGroups: any[],
  onRemove: (id: string, item: any) => void;
  floatRight?: boolean;
  style?: React.CSSProperties;
  classname?: string
}

const Group: React.FC<GroupProp> = (
  {
    floatRight = true,
    selectedGroups,
    onRemove,
    style,
    classname
  }
) => {
  if (!selectedGroups.length) return null;

  const content: any = <List
    itemLayout="horizontal"
    dataSource={selectedGroups}
    renderItem={(item: any) => (
      <List.Item extra={<CloseOutlined className='fr' onClick={() => onRemove(item.id, item)}/>}>
        <List.Item.Meta
          style={{width: 350}}
          avatar={<Avatar src='https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png'/>}
          title={item.title}
          description={item.owner}
        />
      </List.Item>
    )}
  />

  return (
    <DropdownBox floatRight={floatRight}
                 content={content}
                 style={style}
                 classname={classname}>
      已选中群组：{selectedGroups.length}
    </DropdownBox>
  )
}

export default Group;
