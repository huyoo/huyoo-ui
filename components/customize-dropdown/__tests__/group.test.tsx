/**
 * @DECS:
 * @AUTH: hy
 * @DATE: 2021-04-06
 */
import React from "react";
import Group from '../Group'

const list = [
  {
    id: '1',
    title: 'test',
    owner: '121'
  }
];

describe("CustomizeDropdown.type", () => {
  it('should return group dom', function () {
    const dom = <Group onRemove={(id:string)=>{console.log(id)}} selectedGroups={list}/>
    expect(dom).toMatchSnapshot();
  });
})
