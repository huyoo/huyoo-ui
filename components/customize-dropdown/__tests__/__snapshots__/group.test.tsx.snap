// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomizeDropdown.type should return group dom 1`] = `
<Group
  onRemove={[Function]}
  selectedGroups={
    Array [
      Object {
        "id": "1",
        "owner": "121",
        "title": "test",
      },
    ]
  }
/>
`;
